File: PY/10828.py
import sys
N = int(sys.stdin.readline())
result = []
for i in range(N):
    cmd = sys.stdin.readline().split()
    if (cmd[0] == "push"):
        result.append(cmd[1])
    if (cmd[0] == "top"):
        if result:
            print(result[len(result)-1])
        else:
            print(-1)
    if (cmd[0] == "empty"):
        if result:
            print(0)
        else:
            print(1)
    if (cmd[0] == "size"):
        print(len(result))
    if (cmd[0] == "pop"):
        if result:
            print(result.pop())
        else:
            print(-1)

File: CPP/10828.cpp
#include <iostream>
using namespace std;

const int MX = 1000005;
int dat[MX];
int pos = 0;

void push(int x){
    dat[pos++] = x;
}

void pop(){
    if (pos < 1) cout << -1 << '\n';
    else cout << dat[--pos] << '\n';
}

void top(){
    if (pos < 1) cout << -1 << '\n';
    else cout << dat[pos - 1] << '\n';
}

void empty(){
    if (pos == 0) cout << 1 << '\n';
    else cout << 0 << '\n';
}

void size(){
    cout << pos << '\n';
}

int main(void){
    int N;
    cin >> N;
    for (int i = 0; i < N; i++){
        string s;
        int w;
        cin >> s;
        if (s == "push"){
            cin >> w;
            push(w);
        }
        else if (s == "top") top();
        else if (s == "pop") pop();
        else if (s== "empty") empty();
        else if (s == "size") size();
    }
}